### ISSUE TITLE
Parser infinite loop when parsing unsupported match body expression that is surrounded by brackets

### ISSUE THREAD

**Issue by Lubrsi** [SerenityOS/jakt#224]
This causes the parser to go into an infinite loop:
```
function main() {
    match foo {
        (++foo) => 1
    }
}
```

Removing the `()` around `++foo` makes it error:
```
-----
function main() {
    match foo {
        ++foo => 1
    }
}
 
-----
Error: ParserError("expected '=>' after pattern case", Span { file_id: 1, start: 42, end: 44 })
```

**Comment by holychowders** [comment]
Looking into this.

**Comment by lanmonster** [comment]
fixed as of 9012330
